src/pylint_odoo/checkers/odoo_addons.py

```diff
@@ -1197,6 +1204,32 @@ def visit_dict(self, node):
         ):
             self.add_message("manifest-maintainers-list", node=manifest_keys_nodes.get("maintainers") or node)
 
+        # Check there are no external assets
+        if self.linter.is_message_enabled("manifest-external-assets"):
+            assets_node = None
+            for item in node.items:
+                if item[0].value == "assets":
+                    assets_node = item[1]
+
+            # it is important to use the actual astroid.node instead of manifest_dict, otherwise the
+            # errors are not attributed to the proper node.
+            if assets_node:
+                self._check_manifest_external_assets(assets_node)
+
+    def _check_manifest_external_assets(self, node):
+        def is_external_url(url):
+            return urlparse(url).scheme
+
+        for _, item in node.items:
+            for element in item.elts:
+                if isinstance(element, nodes.Const):
+                    if is_external_url(element.value):
+                        self.add_message("manifest-external-assets", node=element)
+                elif isinstance(element, nodes.Tuple):
```

Comment:
Notice it could be a Tuple or a List

See the following example:
 - https://github.com/odoo/odoo/blob/106dc30bebf49335eb7d2450a55d14cd46469e17/addons/barcodes/__manifest__.py#L18

```suggestion
                elif isinstance(element, (nodes.Tuple, nodes.List)):
```

Reply:
In fact, Using manifest_get you could get all the isinstance(..., iterable) object if it is not a string and call it recursively

Reply:
I was checking if it was a tuple because of this use case:

https://github.com/odoo/odoo/blob/032ee36a9556688e2f022ddfb2081800b80efef1/addons/website/__manifest__.py#L241

Odoo docs always show this usage with tuples but I guess in theory you could also use a List although I have not seen it before, I will add it so we cover all possibilities.